### ISSUE TITLE
Remove finalisation step

### ISSUE THREAD

**Issue by mklemarczyk** [animate1978/MB-Lab#172]
**Is your feature request related to a problem? Please describe.**
The feature is not related to any problem. It is request for simplification.
The question is: Is it possible to keep only one state of character?
Why the finalisation operation exists and is it possible to make that operation more transparent?

**Describe the solution you'd like**
Remove finalisation operation. That the options for new created character and finalised character are available all the time. I would like to keep all options available all the time.

**Describe alternatives you've considered**
Clear description of the reason why finalisation operation exists and benefits of having it.

**Additional context**
Current finalisation operation is not very clear why is needed. It disables some options that are no longer available on the MB-Lab character. Moreover, it is not presented anywhere that character is finalised.

**Comment by mklemarczyk** [comment]
Clue 1: Unable to create second non-finalised character.

**Comment by animate1978** [comment]
I like this idea. It has been requested before so I can see this being a feature many people would rejoice to. It would also help people just starting with Blender / MB-Lab kind of ease their way into it, the addon is quite unique and not so user friendly at times, many people try to move their character around before finalizing or try to do other things that break their characters. I've done it myself in the past so I know.

By the way it's a pleasure to have you on board :)

**Comment by mklemarczyk** [comment]
@animate1978 I will start the work than. I expect basic study to finish at beginning of December 2019, since it will be a big project. I would like to create separate branch in main repo, that everybody can try it when it will be possible.

**Comment by mklemarczyk** [comment]
@animate1978 Can you assign me to this issue? I do not have an access to assign myself to issue of change labels.

**Comment by AreebDurrani** [comment]
The issue with this is that while in the creation phase you can only work on one character at a time. Once the characters are finalized MBLab assigns expressions and poses to the one you have selected, but not to the other ones. If this was implemented incorrectly it could lead to not being able to have different expressions and animation on each character in the scene which could make the software unusable for film or animation.

**Comment by TheBricktop** [comment]
So basically it should work like this:
The character holds the data and parameters and editor shows the data of selected character, if You select the other one editor shows the other character parameters accordingly,

**Comment by Anyeos** [comment]
Another faster solution can be having a "revert" button or "make it editable" button to make the character as if it was unfinished. So there you can take care of having only one character per time in editable or creation mode. And I think it can be implemented for now as a workaround meanwhile.

**Comment by mklemarczyk** [comment]
Hello,
I'm sorry for long silence. I started new chapter in my life in France.
I started major refactoring of the interface of the MB-Lab.

Since the development slow down a bit it will be secure for me to release it in chunks.

I expect first chunk with new "create" operator in the end of May 2020.

Stay tuned.

**Comment by animate1978** [comment]
It is quite okay @mklemarczyk I had spent 2 months away myself for personal reasons and am slowly getting back into development of MB-Lab.

**Comment by TheBricktop** [comment]
Its good to see You guys back at it again!

**Comment by mklemarczyk** [comment]
Hello @animate1978 , I tried to integrate changes for long time. I will have to wait or coordinate with issue #234. It is making much changes to the code structure. It would be danger to proceed with both at the same time.

**Comment by animate1978** [comment]
Yes there has been quite a lot of changes to the code for the MB-Dev framework and tools. If removing the finalization feature is not needed at this time there is no need to change.

I agree it would be dangerous at this time to consider.

**Comment by Anyeos** [comment]
Hello guys, I think a good replacement can be a "lock" mode where not all but some things are modificable. I know that the most difficult part is the code, I only say it if someone wants a "finalize" like feature.

Another good feature can be a UI mode (Expert, advanced, simplified) where not all but some options are available. 

But I don't want to distract you or get out of the point.

**Comment by mklemarczyk** [comment]
Hello, I am finally back to Blender. Sorry for so much delay. I would like to create fork for the experimenting in December.
Important question: are there any UI changes ongoing?

**Comment by animate1978** [comment]
> Hello, I am finally back to Blender. Sorry for so much delay. I would like to create fork for the experimenting in December.
> Important question: are there any UI changes ongoing?

No the UI is pretty much done for this version

Welcome back! I too took a long time to get back into this. I just got internet service yesterday at my new place so I will be working on some things and making changes soon, once I get caught up on everything.

**Comment by mklemarczyk** [comment]
Welcome in the new place. I hope you feel well there. I do review what modifications are applicable and what I need to rewrite.
I will move after Christmas too, I hope the internet will be connected from day 1, anyway I can work some time without it. ;)

**Comment by mklemarczyk** [comment]
I started the works, the next version will contain feature I like to call "Fluid UI" and it will not require finalisation step.

**Comment by animate1978** [comment]
Any word on progress related to this?

**Comment by thelabcat** [comment]
😃
